module/zfs/txg.c

```diff
@@ -784,22 +776,23 @@ txg_wait_open(dsl_pool_t *dp, uint64_t txg, boolean_t should_quiesce)
 }
 
 /*
- * If there isn't a txg syncing or in the pipeline, push another txg through
- * the pipeline by quiescing the open txg.
+ * If there isn't a txg quiescing in the pipeline, push the txg
+ * through the pipeline by quiescing the open txg.
+ * It is fine there is a txg still syncing.
  */
 void
-txg_kick(dsl_pool_t *dp)
+txg_kick(dsl_pool_t *dp, uint64_t txg)
 {
 	tx_state_t *tx = &dp->dp_tx;
 
 	ASSERT(!dsl_pool_config_held(dp));
 
 	mutex_enter(&tx->tx_sync_lock);
-	if (!txg_is_syncing(dp) &&
+	txg = txg == 0 ? tx->tx_open_txg : txg;
```

Comment:
```suggestion
	txg = (txg == 0 ? tx->tx_open_txg : txg);
```
nit: Helps the readability a little bit and is consistent with a similar `?:` check in `txg_sync_thread`.

Reply:
Will be fixed in next commit.